hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/KeyOutputStream.java

```diff
@@ -356,6 +370,7 @@ private void handleRetry(IOException exception, long len) throws IOException {
         msg = "Retry request failed. " + action.reason;
         LOG.error(msg, exception);
       }
+      isException = true;
       throw new IOException(msg, exception);
```

Comment:
Lets put 
isException &  throw new IOException(msg, exception); into one function. So that both of these are always updated together.

Reply:
Actually I would say this whole duplication is because RetryPolicy.shouldRetry has two fail states, one when it fails with an exception, and one when it fails with a fail status without an exception by contract.
The handling of this underlying speciality at once should go into some method, and perhaps we should use that method to throw an exception in all the cases, and handle the exception only in this code, and the utility method over RetryPolicy.shouldRetry can do the logging and tinkering with the exception message.
What do you think?

Reply:
Addressed in the latest patch..